code/src/components/app-input-with-icon/AppInputWithIcon.tsx

```diff
@@ -0,0 +1,43 @@
+import { InputBase, InputBaseProps, Paper } from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search";
+import ClearIcon from "@mui/icons-material/Clear";
+import AppIconButton from "@/components/app-icon-button/AppIconButton";
+
+import "@/components/app-input-with-icon/AppInputWithIcon.scss";
+
+type InputWithIconProps = InputBaseProps & {
+  onSearch?: () => void;
+  onClear?: () => void;
+};
```

Comment:
for now we have PR with creation separate file for type. Can create file for that with `.types` move it and import from there. And in other place. don't forget

Reply:
Moved!